assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -0,0 +1,222 @@
+						setIsFetching( false );
+					}
+				}
+			);
+		}
+	};
+
+	useEffect( () => {
+		return () => {
+			isStillMounted = false;
+		};
+	}, [] );
+
+	useEffect( () => {
+		fetchSelectedPost();
+	}, [ postId ] );
+
+	const removePost = () => {
+		setFailedToFetch( false );
+		setAttributes( { postId: null } );
+		setSelectedPost( null );
+	};
+
+	// Since the Page Attachment content is "out of it's borders" then clicking
+	// on the links might sometimes lose focus instead. Using refs for workaround.
+	const removeLinkRef = useRef( null );
+	const closeButtonRef = useRef( null );
+
+	useElementClickDetector( closeButtonRef, () => {
+		toggleAttachment( false );
+	} );
+
+	useElementClickDetector( removeLinkRef, null, true );
+
+	return (
+		<div className="attachment-container">
+			<div className="attachment-wrapper">
+				<div className="attachment-header">
+					{ /* This does not use an IconButton as it replicates the close button on the frontend */ }
+					<span
+						tabIndex="0"
+						className="amp-story-page-attachment-close-button"
+						role="button"
+						onKeyDown={ ( event ) => {
+							if ( ENTER === event.keyCode || SPACE === event.keyCode ) {
+								toggleAttachment( false );
+								event.stopPropagation();
+							}
+						} }
+						ref={ closeButtonRef }
+					/>
+					<RichText
+						value={ title }
+						tagName="div"
+						wrapperClassName="amp-story-page-attachment-title"
+						onChange={ ( value ) => setAttributes( { title: value } ) }
+						placeholder={ __( 'Write Title', 'amp' ) }
+						onClick={ ( event ) => event.stopPropagation() }
+					/>
+					{ postId && (
+						<Button
+							className="remove-attachment-post"
+							onClick={ ( event ) => {
+								event.stopPropagation();
+								removePost();
+							} }
+							isLink
+							isDestructive
+							ref={ removeLinkRef }
+						>
+							{ __( 'Remove Post', 'amp' ) }
+						</Button>
+					) }
+				</div>
+				<div className={ attachmentClass } style={ wrapperStyle }>
+					{ isFetching && <Spinner /> }
+					{ ! isFetching && selectedPost && selectedPost.content && (
+						<RawHTML>
+							{ `<h2>${ selectedPost.title.rendered }</h2>${ selectedPost.content.rendered }` }
+						</RawHTML>
+					) }
+					{ ! isFetching && ( ! postId || failedToFetch ) && (
+						<>
+							{ failedToFetch && (
+								<div>
+									<p className="failed-message">
+										{ __( 'The selected post failed to load, please select a new post or try loading again.', 'amp' ) }
+										<Button
+											className="refetch-attachment-post"
+											onClick={ ( event ) => {
+												event.stopPropagation();
+												fetchSelectedPost();
+											} }
+											isLink
+										>
+											{ __( ' Try again', 'amp' ) }
+										</Button>
+									</p>
+								</div>
+							) }
+							<PostSelector
+								labelText={ __( 'Attachment Content', 'amp' ) }
+								placeholder={ __( 'Search for a post or page ...', 'amp' ) }
```

Comment:
Make this more generic, in case someone has filtered the support post types to include custom post types.